File: application/src/components/Forms/src/com/eas/client/forms/api/HorizontalPosition.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.eas.client.forms.api;

import com.eas.script.AlreadyPublishedException;
import com.eas.script.HasPublished;
import com.eas.script.NoPublisherException;
import com.eas.script.ScriptFunction;
import com.eas.script.ScriptObj;
import jdk.nashorn.api.scripting.JSObject;

/**
 *
 * @author mg
 */
@ScriptObj(jsDoc = "/**\n"
        + "* Horizontal position constants.\n"
        + "*/")
public class HorizontalPosition implements HasPublished {

    public static final int LEFT = 2;
    public static final int CENTER = 0;
    public static final int RIGHT = 4;
    
    private static JSObject publisher;
    protected Object published;
    
    @ScriptFunction
    public static int getLEFT() {
        return LEFT;
    }
    
    @ScriptFunction
    public static int getCENTER() {
        return CENTER;
    }
    
    @ScriptFunction
    public static int getRIGHT() {
        return RIGHT;
    }
    
    @Override
    public Object getPublished() {
        if (published == null) {
            if (publisher == null || !publisher.isFunction()) {
                throw new NoPublisherException();
            }
            published = publisher.call(null, new Object[]{this});
        }
        return published;
    }

    @Override
    public void setPublished(Object aValue) {
        if (published != null) {
            throw new AlreadyPublishedException();
        }
        published = aValue;
    }

    public static void setPublisher(JSObject aPublisher) {
        publisher = aPublisher;
    }
}

File: application/api/forms/horizontal-position.js
(function() {
    var javaClass = Java.type("com.eas.client.forms.api.HorizontalPosition");
    javaClass.setPublisher(function(aDelegate) {
        return new P.HorizontalPosition(aDelegate);
    });
    
    /**
     * Generated constructor.
     * @constructor HorizontalPosition HorizontalPosition
     */
    P.HorizontalPosition = function () {

        var maxArgs = 0;
        var delegate = arguments.length > maxArgs ?
            arguments[maxArgs] : new javaClass();

        Object.defineProperty(this, "unwrap", {
            get: function() {
                return function() {
                    return delegate;
                };
            }
        });
        /**
         * Generated property jsDoc.
         * @property CENTER
         * @memberOf HorizontalPosition
         */
        Object.defineProperty(this, "CENTER", {
            get: function() {
                var value = delegate.CENTER;
                return P.boxAsJs(value);
            }
        });

        /**
         * Generated property jsDoc.
         * @property LEFT
         * @memberOf HorizontalPosition
         */
        Object.defineProperty(this, "LEFT", {
            get: function() {
                var value = delegate.LEFT;
                return P.boxAsJs(value);
            }
        });

        /**
         * Generated property jsDoc.
         * @property RIGHT
         * @memberOf HorizontalPosition
         */
        Object.defineProperty(this, "RIGHT", {
            get: function() {
                var value = delegate.RIGHT;
                return P.boxAsJs(value);
            }
        });


        delegate.setPublished(this);
    };
})();